docs/rfc/block-registration.md

```diff
@@ -69,19 +69,32 @@ To register a new block type, start by creating a `block.json` file. This file:
 	"icon": "star",
 	"description": "Shows warning, error or success notices  ...",
 	"keywords": [ "alert", "message" ],
+	"textDomain": "my-plugin",
 	"attributes": {
 		"message": {
 			"type": "string",
 			"source": "html",
 			"selector": ".meessage"
 		}
 	},
-	"edit": "blocks/notice-edit.js",
-	"save": "blocks/notice-save.js",
 	"styles": [ 
 		{ "name": "default", "label": "Default", "isDefault": true }, 
 		{ "name": "other", "label": "Other" }
-	]
+	],
+	"editorScript": "file:build/editor.assets.json",
+	"script": {
+		"handle": "my-plugin-notice",
+		"file": "index.js",
+		"dependencies": [ "wp-element", "wp-components" ],
```

Comment:
Based on the discussion on this ticket https://core.trac.wordpress.org/ticket/47350 I'd think that omitting the "dependencies" here would automatically mean, check for the existence of `index.deps.json` to load them.

Reply:
We are going to deprecate `index.deps.json` in favor of `index.asset.json` which will contain the full object instead. It doesn't make sense to go so granular in my opinion.

Reply:
What's the reason for this change?

Reply:
You can have better control over `version` with webpack as you can use the hash it generates based on the content of the file. Webpack knows the path of file as well. It also can generate the handle so why to even bother with it manually?

Reply:
I'm thinking from a SoC Point of view. We shouldn't be designing this `block.json` API based on the fact that we have a wepack plugin that can generate some of it. Some people might not use webpack, others might. So what's the common best API for both